adr/ADR-0003-trustyai-service-deployment-using-operator-pattern.md

```diff
@@ -0,0 +1,245 @@
+---
+num: 3 # allocate an id when the draft is created
+title: TrustyAI Service Deployment using Operator pattern
+status: "Draft" # One of Draft, Accepted, Rejected
+authors:
+  - "ruivieira"
+  - "danielezonca" # One item for each author, as github id or "firstname lastname"
+tags:
+  - "service" # e.g. service, python, java, etc
+---
+
+## Title
+
+TrustyAI Service Deployment using Operator pattern
+
+## Context and Problem Statement
+
+The [TrustyAI Service](https://github.com/trustyai-explainability/trustyai-explainability/tree/main/explainability-service) can be deployed manually as a standalone container or via [ODH-manifest](https://github.com/opendatahub-io/odh-manifests/blob/master/trustyai-service/) as part of ODH KfDef. Both cases have limitations: a plain Deployment is error prone for the users (some parameters are mandatory) and the ODH-manifest contains [some hacks](https://github.com/opendatahub-io/odh-manifests/blob/master/trustyai-service/default/trustyai-deployment.yaml#L69-L87).
+A Kubernetes operator would provide a simple and consistent way to deploy and manage the TrustyAI service.
+
+In addition to this, the deployment and the storage (PVC for now) must be created into a user owned namespace to give users full control and prevent security issues. An operator can enforce this.
+
+## Goals
```

Comment:
We should document OperatorHub integration as a non-goal for now, but something that may be done at a later date.